e2e/testcases/helm_sync_test.go

```diff
@@ -380,3 +409,75 @@ func helmChartVersion(chartVersion string) nomostest.Sha1Func {
 		return chartVersion, nil
 	}
 }
+
+// pushes a new helm chart for use during an e2e test
+func pushHelmChart(nt *nomostest.NT, t *testing.T, helmchart, version string) string {
+	t.Helper()
+
+	nt.T.Log("Push helm chart to the artifact registry")
+	tmpDir := t.TempDir()
+
+	chartName := fmt.Sprintf("%s-%s-%s-%s", helmchart, nomostesting.GCPProjectIDFromEnv, nomostesting.GCPClusterFromEnv, generateRandomString())
+	if len(chartName) > 50 {
+		// the chartName + releaseName is used as the metadata.name of resources in the coredns helm chart, so we must trim this down
+		// to keep it under the k8s length limit
+		chartName = chartName[len(chartName)-50:]
+		chartName = strings.Trim(chartName, "-")
+	}
+
+	paths, err := filepath.Glob(fmt.Sprintf("../testdata/helm-charts/%s/*", helmchart))
+	if err != nil {
+		nt.T.Fatalf("failed to get chart filepaths: %v", err)
+	}
+	for _, path := range paths {
+		if _, err := nt.Shell.Command("cp", "-r", path, tmpDir).CombinedOutput(); err != nil {
```

Comment:
I think for these filesystem manipulations (e.g. `cp` and `sed`) it would be preferred to use a pure Go alternative rather than shelling out to another binary

Reply:
done